src/test/java/org/opensearch/security/dlic/rest/api/UserApiTest.java

```diff
@@ -75,6 +75,26 @@ public void testSecurityRoles() throws Exception {
         checkGeneralAccess(HttpStatus.SC_OK, "newuser", "newuser");
     }
 
+    @Test
+    public void testParallelPutRequests() throws Exception {
+        setup();
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+
+        HttpResponse[] responses = rh.executeMultipleAsyncPutRequest(10, "/_opendistro/_security/api/internalusers/test1", "{\"password\":\"test1\"}");
```

Comment:
Must test both legacy and opensearch endpoints

Reply:
done